### ISSUE TITLE
How does this compare to other libraries with compiled with AVX instructions

### ISSUE THREAD

**Issue by gytis-ivaskevicius** [bytedance/sonic#153]
So I am sure that you guys were not like 'welp, I heard assembly is fast, let's use that!' but instead did some benchmarking of other speedy JSON libraries. 

So the case I am interested in is:
JSON library that is written in C/C++/Rust/Zig library AND compiled with AVX instructions enabled. 
How does it compare? Do you guys have/remember the results of that?

**Comment by chenzhuoyu** [comment]
> So I am sure that you guys were not like 'welp, I heard assembly is fast, let's use that!' but instead did some benchmarking of other speedy JSON libraries.

We did.

> How does it compare? Do you guys have/remember the results of that?

TL;DR: it depends.

Sometimes Sonic is faster than every other library even they are written in C++ or Rust, sometimes not.

Libraries that written in languages compiled by `gcc` or `llvm` tends to be faster than those written in `Go` on average. It's not fair to compare apples with oranges, language and compiler differences are larger than you might think.
